content/concepts/protocols.md

```diff
@@ -1,227 +1,167 @@
-properties:
+The "medium" over which a libp2p protocol transpires is a bi-directional binary stream 
+with the following properties:
 
 - Bidirectional, reliable delivery of binary data
   - Each side can read and write from the stream at any time
   - Data is read in the same order as it was written
-  - Can be "half-closed", that is, closed for writing and open for reading, or closed for reading and open for writing
+  - Can be "half-closed", that is, closed for writing and open for reading, or closed 
+    for reading and open for writing
 - Supports backpressure
-  - Readers can't be flooded by eager writers <!-- TODO(yusef) elaborate: how is backpressure implemented? is it transport-depdendent? -->
+  - Eager writers can't flood readers
 
-Behind the scenes, libp2p will also ensure that the stream is [secure](/concepts/secure-comms/) and efficiently
-[multiplexed](/concepts/stream-multiplexing/). This is transparent to the protocol handler, which reads and writes
-unencrypted binary data over the stream.
-
-The format of the binary data and the mechanics of what to send when and by whom are all up to the protocol to determine. For inspiration, some [common patterns](#common-patterns) that are used in libp2p's internal protocols are outlined below.
+Behind the scenes, libp2p will also ensure that the stream is 
+[secure](/concepts/secure-comms/) and efficiently
+[multiplexed](/concepts/stream-multiplexing/). This is transparent to the protocol 
+handler, which reads and writes unencrypted binary data over the stream.
 
+The protocol determines the binary data format and the transport mechanics. 
+For inspiration, some [common patterns](#common-patterns) that are used in libp2p's 
+internal protocols are outlined below.
 
 ## Protocol Negotiation
 
-When dialing out to initiate a new stream, libp2p will send the protocol id of the protocol you want to use.
-The listening peer on the other end will check the incoming protocol id against the registered protocol handlers.
-
-If the listening peer does not support the requested protocol, it will end the stream, and the dialing peer can
-try again with a different protocol, or possibly a fallback version of the initially requested protocol.
-
-If the protocol is supported, the listening peer will echo back the protocol id as a signal that future data
-sent over the stream will
-use the agreed protocol semantics.
-
-This process of reaching agreement about what protocol to use for a given stream or connection is called
-**protocol negotiation**.
-
+When dialing out to initiate a new stream, libp2p will send the protocol ID of the 
+protocol you want to use. The listening peer on the other end will check the incoming 
+protocol ID against the registered protocol handlers.
 
-### Matching protocol ids and versions
+If the listening peer does not support the requested protocol, it will end the stream, 
```

Comment:
Not sure if we want to go into multistream specifics in this doc, or if we have another doc for that. Either way, it doesn't "end" the stream (that sounds like the stream is getting closed, which it isn't).

```suggestion
If the listening peer does not support the requested protocol, it will send "na" on the stream
```